lib/sidekiq/middleware/chain.rb

```diff
@@ -118,7 +117,7 @@ def exists?(klass)
       end
 
       def empty?
-        @entries.nil? || @entries.empty?
+        !instance_variable_defined?(:@entries) || entries.empty?
```

Comment:
You can still keep this as `@entries.nil?`. An undefined instance variable is nil. Also, if the variable _is_ defined, but is nil, this method will return `true`.

Reply:
I tried that originally, but there was a bunch of log spew:
```
sidekiq/lib/sidekiq/middleware/chain.rb:121: warning: instance variable @entries not initialized
```

I agree someone was likely trying to avoid allocating the array unless needed (which seems right).  this certainly isn't as clean, so happy to revert this part of it